py_feels/binary/training/binary_classifier_training.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Dec  7 15:45:21 2017
+
+Training script for binary text classifier
+
+@author: rcehemann
+"""
+###############################################################################
+
+# logistical stuff
+import pandas as pd                                     # pandas allows easy manipulation of the data
+from sklearn.feature_extraction.text import TfidfTransformer, CountVectorizer # preprocessors
+from sklearn.feature_selection import chi2, SelectKBest # used to filter useless data
+from sklearn.pipeline import Pipeline                   # pre-processing pipeline
+from sklearn.model_selection import cross_val_score     # for testing accuracy
+from sklearn.externals.joblib import dump               # fitted model will be pickled for later use
+
+# classification algorithms
+from sklearn.svm      import SVC                        # support-vector machine classifier
+from sklearn.ensemble import RandomForestClassifier, GradientBoostingClassifier # tree-based classifiers
+from sklearn.neural_network import MLPClassifier        # multilayer perceptron classifier (dense NN)
+from sklearn.naive_bayes import MultinomialNB           # naive bayes classifier w/ multinomial distribution
+
+###############################################################################
+
+
+# assemble training data from various files into pandas DataFrame objects.
+# format of the files is a sentence followed by a sentiment
+# 1 == positive sentiment and 0 == negative sentiment
+amazon = pd.read_csv('UCI_training_data/amazon_cells_labelled.txt', 
+                     sep="\t", 
+                     names=['sentence', 'sentiment'], 
+                     dtype={'sentence':str, 'sentiment':int})
+yelp   = pd.read_csv('UCI_training_data/yelp_labelled.txt', 
+                     sep="\t", 
+                     names=['sentence', 'sentiment'], 
+                     dtype={'sentence':str, 'sentiment':int})
+imdb   = pd.read_csv('UCI_training_data/imdb_labelled.txt', 
+                     sep="\t", 
+                     names=['sentence', 'sentiment'], 
+                     dtype={'sentence':str, 'sentiment':int})
+
+# concatenate the individual dataframes, shuffle them (sample with frac=1) and reset the index
+data   = pd.concat([amazon, yelp, imdb], ignore_index = True).sample(frac=1).reset_index(drop=True)
```

Comment:
There are some really long lines in this file. This line should be the widest. 100 ruler width is a nice place to aim.

Reply:
I'll split the long ones.  WRT the old directory, it's not on my local copy and i'm not sure why it appears on GH. I'll add the .DS_store files to .gitignore